gitpod-io/gitpod
# Use prettier for TypeScript files in .werft

## Description


This PR introduces automatic formatting of all TypeScript files in the `.werft` folder. 

1. It configures [prettier](https://prettier.io/) to perform automatic formatting on all TypeScript files in the `.werft` folder as part of our pre-commit hooks.
2. It adds `esbenp.prettier-vscode` to the list of extensions
3. It configures VSCode to fun "format on save" on TypeScript files and sets prettier as the formatter

This PR also includes performing the automatic formatting, hence the rather large diff. For the actual code review I suggest you focus on [43ca5ee](https://github.com/gitpod-io/gitpod/pull/8245/commits/43ca5ee418fbdf15433134dce9fbbc579d7d1bcb).

## Related Issue(s)

Fixes https://github.com/gitpod-io/gitpod/issues/8066

## How to test


Verify that it performs formatting when you run

```
pre-commit run --all-files --show-diff-on-failure
```

Edit a `ts` file in `server` and verify it doesn't use prettier to format the file when saving (that is, the prettier extension that is now installed as part of the workspace correctly ignores the entire components folder)

## Release Notes

```release-note
NONE
```

## Documentation

N/A
